utils/common/src/main/java/org/apache/brooklyn/util/javalang/MethodAccessibleReflections.java

```diff
@@ -0,0 +1,142 @@
+        } catch (SecurityException e) {
+            boolean added = SET_ACCESSIBLE_FAILED_LOGGED_METHODS.add(method.toString());
+            if (added) {
+                LOG.warn("Problem setting accessible for method " + method, e);
+            } else {
+                if (LOG.isTraceEnabled()) LOG.trace("Problem setting accessible for method " + method, e);
+            }
+            return false;
+        }
+    }
+
+    /**
+     * @see {@link Reflections#findAccessibleMethod(Method)}
+     */
+    static Method findAccessibleMethod(Method method) {
+        if (!Modifier.isPublic(method.getModifiers())) {
+            trySetAccessible(method);
+            return method;
+        }
+        boolean declaringClassPublic = Modifier.isPublic(method.getDeclaringClass().getModifiers());
+        if (!declaringClassPublic) {
+            // reflectively calling a public method on a private class can fail, unless we first set it
+            // call setAccessible. But first see if there is a public method on a public super-type
+            // that we can call instead!
+            Maybe<Method> publicMethod = tryFindPublicEquivalent(method);
+            if (publicMethod.isPresent()) {
+                LOG.debug("Switched method for publicly accessible equivalent: method={}; origMethod={}", publicMethod.get(), method);
+                return publicMethod.get();
+            } else {
+                trySetAccessible(method);
+                return method;
+            }
+        }
+        
+        return method;
+    }
+    
+    private static Maybe<Method> tryFindPublicEquivalent(Method method) {
+        if (Modifier.isStatic(method.getModifiers())) {
+            return Maybe.absent();
+        }
+        
+        Class<?> clazz = method.getDeclaringClass();
+        
+        for (Class<?> interf : clazz.getInterfaces()) {
+            Maybe<Method> altMethod = tryFindPublicMethod(interf, method.getName(), method.getParameterTypes());
+            if (altMethod.isPresent()) {
+                return altMethod;
+            }
+        }
+        
+        Class<?> superClazz = clazz.getSuperclass();
+        while (superClazz != null) {
+            Maybe<Method> altMethod = tryFindPublicMethod(superClazz, method.getName(), method.getParameterTypes());
+            if (altMethod.isPresent()) {
+                return altMethod;
+            }
+            superClazz = superClazz.getSuperclass();
+        }
+        
+        return Maybe.absent();
+    }
+
+    private static Maybe<Method> tryFindPublicMethod(Class<?> clazz, String methodName, Class<?>... parameterTypes) {
+        if (!Modifier.isPublic(clazz.getModifiers())) {
+            return Maybe.absent();
+        }
+        
+        try {
+            Method altMethod = clazz.getMethod(methodName, parameterTypes);
+            if (Modifier.isPublic(altMethod.getModifiers()) && !Modifier.isStatic(altMethod.getModifiers())) {
+                return Maybe.of(altMethod);
```

Comment:
Skip it if the class is not accessible. This will fail with same problem if you have:
public class 1 - public method
protected class 2 < class 1 - public method
protected class 3 < class 2 - public method